Clean unnecessary auto generated files

AlonaKaplan opened the pull request:
Remove prometheus, RBAC proxy, scorecard, leader election, webhook, cretificate manager - as currently the project is minimal and those advanced options are not required.

Also, remove a binary that was merged to the repo by mistake.

AlonaKaplan committed:
Remove prometheus

The project doesn't need a metrics-monitor at this point.

Signed-off-by: Alona Paz <alkaplan@redhat.com>

AlonaKaplan committed:
Remove RBAC proxy

An RBAC proxy is not requried at this stage of the project.

Signed-off-by: Alona Paz <alkaplan@redhat.com>

AlonaKaplan committed:
Remove scorecard

The project doesn't require it atm.

Signed-off-by: Alona Paz <alkaplan@redhat.com>

AlonaKaplan committed:
Don't include binaries and idea related files in git

Signed-off-by: Alona Paz <alkaplan@redhat.com>

AlonaKaplan committed:
Remove leader election

The Deployment has only one instance atm, so leader election is not
required.

Signed-off-by: Alona Paz <alkaplan@redhat.com>

AlonaKaplan committed:
Remove webhook

The CRD doesn't require webhook atm.

Signed-off-by: Alona Paz <alkaplan@redhat.com>

AlonaKaplan committed:
Remove cretificate manager

It is not required atm.

Signed-off-by: Alona Paz <alkaplan@redhat.com>

ormergi commented on main.go:
```diff
@@ -67,7 +65,7 @@ func main() {
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
 		Scheme:                 scheme,
-		MetricsBindAddress:     metricsAddr,
+		MetricsBindAddress:     "0", // disable metrics
```
If it possible, lets remove this field

AlonaKaplan replied:
I don't think it is possible. The default is 8080.
```
// DefaultBindAddress sets the default bind address for the metrics listener
// The metrics is on by default.
var DefaultBindAddress = ":8080"
```

ormergi replied:
Nice :slightly_smiling_face:

ormergi commented on config/default/kustomization.yaml:
```diff
@@ -1,63 +0,0 @@
-# Adds namespace to all resources.
```
This file configures all related yaml manifests of this project operator, including CRDs, RBAC objects and Deployment,
It will patch the namespace and all objects names with the names specified in this file (namespace - line 2, name-prefix - line 9).

Please keep this file and remove the commented lines related to webhook and certmanager.

AlonaKaplan replied:
done.

ormergi reviewed (commented):
I think its also possible to remove healthz and readys probs, they were required for auth-proxy container that used for metrics endpoint that been also removed.
Although its can come in handy if other services need to relay on the subject controller readiness.

ormergi commented:
Makefile install the project tooling if they are not exist (such as controller-gen, Kustomize) under the project rook `./bin`, could you also add `./bin/` to .gitignore?

AlonaKaplan commented:
> I think its also possible to remove healthz and readys probs, they were required for auth-proxy container that used for metrics endpoint that been also removed. Although its can come in handy if other services need to relay on the subject controller readiness.

I can remove it. But why do you think it is related to the metrics? It is just a simple ping.

ormergi commented:
> > I think its also possible to remove healthz and readys probs, they were required for auth-proxy container that used for metrics endpoint that been also removed. Although its can come in handy if other services need to relay on the subject controller readiness.
> 
> I can remove it. But why do you think it is related to the metrics? It is just a simple ping.

There is a patch manifest that adds the auth-proxy container to the subject controller Deployment manifest that specify the healthz (and metrics endpoint):
https://github.com/AlonaKaplan/selfserviceoverlay/blob/08992981ee0b46eb0bff60350a07a264084f5eea/config/default/manager_auth_proxy_patch.yaml#L53-L54
I guess its to start when the controller manager is ready.

ormergi commented on config/default/manager_auth_proxy_patch.yaml:
```diff
@@ -1,54 +0,0 @@
-# This patch inject a sidecar container which is a HTTP proxy for the
```
I guess this change related to previous commit.

If you get a chance to rebase again it will be nice moving it to the related commit, not a must.

AlonaKaplan replied:
Why? It is the RBAC proxy, exactly what this commit is removing. What do I miss?

ormergi commented on .gitignore:
```diff
@@ -7,6 +7,7 @@
 *.dll
 *.so
 *.dylib
+bin/
```
nit: It will be nice adding a comment saying this directory related to the project tooling binaries, not a must.

AlonaKaplan replied:
done.

ormergi commented on .gitignore:
```diff
@@ -19,3 +20,5 @@
 # Go workspace file
 go.work
+
+.idea/
```
nit: since the project doesnt introduce this dir it should be configured in the  local env .gitignore.

AlonaKaplan replied:
what is local env .gitignore?

ormergi replied:
Its similar to the project gitignore but affects whole local environment, here is a nice gist that explain what is it and  how to work with it https://gist.github.com/subfuzion/db7f57fff2fb6998a16c

ormergi reviewed (approved):
LGTM

AlonaKaplan commented:
> > > I think its also possible to remove healthz and readys probs, they were required for auth-proxy container that used for metrics endpoint that been also removed. Although its can come in handy if other services need to relay on the subject controller readiness.
> > 
> > 
> > I can remove it. But why do you think it is related to the metrics? It is just a simple ping.
> 
> There is a patch manifest that adds the auth-proxy container to the subject controller Deployment manifest that specify the healthz (and metrics endpoint):
> 
> https://github.com/AlonaKaplan/selfserviceoverlay/blob/08992981ee0b46eb0bff60350a07a264084f5eea/config/default/manager_auth_proxy_patch.yaml#L53-L54
> 
> I guess its to start when the controller manager is ready.

The parameter was on the manager controller and not on the kube-proxy one. I move it to the general Deployment yaml.
And added a commit to remove the health check. We don't need it for now, we can re-introduce it in the future if needed.

AlonaKaplan committed:
Remove liveness and readiness probes

It is not needed atm.

Signed-off-by: Alona Paz <alkaplan@redhat.com>

ormergi commented:
> The parameter was on the manager controller and not on the kube-proxy one. I move it to the general Deployment yaml.

This manifest is a patch that add the kube-proxy cotnainer with the mentioned parameters, eventually it become part of  to the controller Deployment by Kustomize.
It doesnt seem to add the mentioned parameter to the controller container.

ormergi commented:
LGTM

AlonaKaplan commented:
> > The parameter was on the manager controller and not on the kube-proxy one. I move it to the general Deployment yaml.
> 
> This manifest is a patch that add the kube-proxy cotnainer with the mentioned parameters, eventually it become part of to the controller Deployment by Kustomize. It doesnt seem to add the mentioned parameter to the controller container.

https://github.com/AlonaKaplan/selfserviceoverlay/blob/08992981ee0b46eb0bff60350a07a264084f5eea/config/default/manager_auth_proxy_patch.yaml#L51 - you can see here iti s added to the manager container and not to the kubeproxy one.
